### DISCUSSION TITLE
Exercise 6, question 3

### DISCUSSION THREAD

**Discussion by JordanTrippett** [LiNk-NY/IntroR#41] [Q&A] [answered]
Hey all,
I'm having trouble loading the dataset "babies.txt" in exercise 6 on question 3. I just keep getting the output:

_Warning message in data(babies.txt):
“data set ‘babies.txt’ not found”_

Does anyone have any tips?
Thanks in advance!

**Comment by ralythompson24** [comment, marked as answer]
Hi @JordanTrippett, sorry for the confusion but the `babies.txt` dataset is a file you will have to import into your google colab session from an actual text file. The text file is saved on Blackboard under Weekly Materials > Datasets. 

The dataset needs to be uploaded into your google drive in a folder that can be seen by google colab. On the left hand side of the console, there's an icon that looks like a file with an upward pointing arrow. That's where you can upload `babies.txt`.

Note that you will also need to load the data into the R session as a `data.frame`. Recall the `read.table()` function and note that the txt file has headers.

**Reply by JordanTrippett** [reply]
Thank you so much!

**Reply by emilyyli** [reply]
<img width="1406" alt="image" src="https://github.com/CUNYSPHCode/IntroR/assets/171090659/f5f5da99-d4e3-4bbc-ab75-7f370b901e66">
@ralythompson24 I followed the steps you mentioned above but am still having trouble loading the data

**Reply by ralythompson24** [reply]
Hi @emilyyli so you actually need to use the `read.table()` function to read in the txt file and create your `data.frame` object. https://www.statology.org/r-read-table/

**Reply by emilyyli** [reply]
@ralythompson24 I loaded the txt file and was able to get my code running. However if I exit the google colab session or restart it, the code won't run unless I redownload the txt file again in the top left corner (where the up arrow is with the file) because with each new session, the "babies.txt" file is gone. Is it supposed to be like this? 


<img width="1427" alt="image" src="https://github.com/CUNYSPHCode/IntroR/assets/171090659/c2b19b90-4e34-448f-bbc1-ca098cca77f9">

**Reply by ralythompson24** [reply]
That's strange... is your .ipynb file saved in your google drive? I think if you don't save things to your g-drive google colab will delete data files, etc. when you close the session.

**Reply by emilyyli** [reply]
Hi @ralythompson24 I tried saving it as a copy onto my google drive but I'm still facing the same problem.. 
<img width="1220" alt="image" src="https://github.com/CUNYSPHCode/IntroR/assets/171090659/97f35a57-4f3e-4c6f-97d0-d9da15c6a912">

**Reply by LiNk-NY** [reply]
The runtime is temporary and you should expect the data saved on to it to be deleted. 
Have you tried loading the data from your Google Drive? 
The data you show here seems to be saved in the runtime environment and not the GDrive folder.

**Reply by emilyyli** [reply]
I've tried to include a link to the "babies.txt" file on my own Google Drive but it doesn't load

**Reply by LiNk-NY** [reply]
Ah, it looks like you can only mount your Google Drive for Python runtimes :/

**Reply by ralythompson24** [reply]
@LiNk-NY I just discovered this too:
<img width="556" alt="image" src="https://github.com/CUNYSPHCode/IntroR/assets/64804055/a4291726-5430-4d10-96e4-cd74ad73e8c3">

**Reply by LiNk-NY** [reply]
That means you have to upload your data every time you work on the colab notebook.

FWIW, it may be possible to load from a GitHub repository but I think that 
requires the notebook to be in the same repository.

**Reply by ralythompson24** [reply]
Sorry @emilyyli it looks like you'll have to re-upload the data every time you start a new session in Google Colab.

Later this week we will be getting ourselves set up in RStudio. In that environment, you'll be able to save data in your computer's drive and access it that way (this is the typical workflow for using R, if you continue to use it beyond this class, which I hope you do!)

**Reply by emxlii** [reply]
@ralythompson24 Will this affect grading when you download my ipyn file? Since it means it wont load the plot or dataframe table for the last question too

**Reply by ralythompson24** [reply]
No, it won't affect the grading. I don't even usually run the code when I grade (unless I see something that I think won't run).
